lib/rust/parser/debug/src/lib.rs

```diff
@@ -0,0 +1,196 @@
+    //  type info; then filter by flag here instead of hard-coding these simplifications.
+    let token_to_str = move |token: Value| {
+        let range = token_code_range(&token, base);
+        code[range].to_owned().into_boxed_str()
+    };
+    let token_to_str_ = token_to_str.clone();
+    to_s_expr.mapper(ident_token, move |token| Value::symbol(token_to_str_(token)));
+    let token_to_str_ = token_to_str.clone();
+    to_s_expr.mapper(operator_token, move |token| Value::string(token_to_str_(token)));
+    let token_to_str_ = token_to_str.clone();
+    to_s_expr.mapper(text_section_token, move |token| Value::string(token_to_str_(token)));
+    let token_to_str_ = token_to_str.clone();
+    to_s_expr.mapper(number_token, move |token| Value::string(token_to_str_(token)));
+    let token_to_str_ = token_to_str;
+    to_s_expr.mapper(number_base_token, move |token| Value::string(token_to_str_(token)));
+    let into_car = |cons| match cons {
+        Value::Cons(cons) => cons.into_pair().0,
+        _ => panic!(),
+    };
+    let simplify_case = |list| {
+        let list = strip_hidden_fields(list);
+        let (_, list) = match list {
+            Value::Cons(cons) => cons.into_pair(),
+            _ => panic!(),
+        };
+        let (expression, list) = match list {
+            Value::Cons(cons) => cons.into_pair(),
+            _ => panic!(),
+        };
+        let (_, list) = match list {
+            Value::Cons(cons) => cons.into_pair(),
+            _ => panic!(),
+        };
+        Value::cons(expression, list)
+    };
+    let simplify_escape = |mut list| {
+        let mut last = None;
+        while let Value::Cons(cons) = list {
+            let (car, cdr) = cons.into_pair();
+            last = Some(car);
+            list = cdr;
+        }
+        last.unwrap()
+    };
+    let strip_invalid = |list| {
+        let Value::Cons(cons) = list else { unreachable!() };
+        let (car, _) = cons.into_pair();
+        Value::cons(car, Value::Null)
+    };
+    let line = rust_to_meta[&tree::block::Line::reflect().id];
+    let operator_line = rust_to_meta[&tree::block::OperatorLine::reflect().id];
+    let case = rust_to_meta[&tree::CaseOf::reflect().id];
+    let invalid = rust_to_meta[&tree::Invalid::reflect().id];
+    to_s_expr.mapper(line, into_car);
+    to_s_expr.mapper(operator_line, into_car);
+    to_s_expr.mapper(case, simplify_case);
+    to_s_expr.mapper(invalid, strip_invalid);
+    to_s_expr.mapper(text_escape_token, simplify_escape);
+    to_s_expr.skip(newline_token);
+    to_s_expr.skip(wildcard_token);
+    to_s_expr.skip(autoscope_token);
+    to_s_expr.skip(text_start_token);
+    to_s_expr.skip(text_end_token);
+    to_s_expr.skip(open_symbol_token);
+    to_s_expr.skip(close_symbol_token);
+    tuplify(to_s_expr.value(ast_ty, &value))
+}
+
+/// Strip certain fields that should be excluded from output.
```

Comment:
What do you mean by "should be excluded from output"? These docs do not explain what this function does.